addons/mail/static/src/core/web/activity_service.js

```diff
@@ -2,25 +2,31 @@
 
 import { _t } from "@web/core/l10n/translation";
 import { registry } from "@web/core/registry";
-import { browser } from "@web/core/browser/browser";
 
 export class ActivityService {
     /**
      * @param {import("@web/env").OdooEnv} env
      * @param {Partial<import("services").Services>} services
      */
     constructor(env, services) {
-        try {
-            // useful for synchronizing activity data between multiple tabs
-            this.broadcastChannel = new browser.BroadcastChannel("mail.activity.channel");
-            this.broadcastChannel.onmessage = this._onBroadcastChannelMessage.bind(this);
-        } catch {
-            // BroadcastChannel API is not supported (e.g. Safari < 15.4), so disabling it.
-            this.broadcastChannel = null;
-        }
         this.env = env;
         this.store = services["mail.store"];
         this.orm = services.orm;
+        this.multiTab = services["multi_tab"];
+        this.multiTab.bus.addEventListener("mail.activity.insert", ({ detail }) => {
+            this.store.Activity.insert(detail, { broadcast: false });
+        });
+        this.multiTab.bus.addEventListener("mail.activity.delete", ({ detail }) => {
+            const activity = this.store.Activity.insert(detail, { broadcast: false });
+            this.delete(activity, { broadcast: false });
+        });
+        this.multiTab.bus.addEventListener("mail.activity.reload", ({ detail }) => {
+            const thread = this.env.services["mail.thread"].getThread(
+                detail.resModel,
+                detail.resId
```

Comment:
I would keep event name to `serviceName/action`, e.g. `mail.activity/insert`

Reply:
Renamed the events.